Kudu.Core/Settings/DiagnosticsSettingsManager.cs

```diff
@@ -69,6 +69,12 @@ private DiagnosticsSettings ReadSettings()
                 {
                     string fileContent = null;
                     OperationManager.Attempt(() => fileContent = FileSystemHelpers.ReadAllTextFromFile(_path));
+                    if (DiagnosticsSettings.AutoCorrect(_tracer, fileContent, out fileContent))
+                    {
+                        // if there is corrupted value, reset it back to file system
+                        FileSystemHelpers.WriteAllTextToFile(_path, fileContent);
+                    }
+
                     return JsonConvert.DeserializeObject<DiagnosticsSettings>(fileContent);
```

Comment:
the reason i have auto correction is, there are corrupted data exist in some user`s app. we would like to automatically fix those broken data.

for number value, serialization won`t cache issue.
````
           var result = (TraceEventType)2;
           Console.WriteLine(result);                      // will print "Error"
````

the error we see from controller, the XmlWriter must have a customer XML serializer to make sure value is define in enum